test/matrix_board_test.exs

```diff
@@ -0,0 +1,125 @@
+defmodule MatrixBoardTest do
+  use ExUnit.Case
+
+  describe "creating a board" do
+    test "3x3 board has 9 spaces" do
+      assert [[nil, nil, nil], [nil, nil, nil], [nil, nil, nil]] == MatrixBoard.create_board(3)
+    end
+
+    test "4x4 board has 16 spaces" do
+      assert [
+               [nil, nil, nil, nil],
+               [nil, nil, nil, nil],
+               [nil, nil, nil, nil],
+               [nil, nil, nil, nil]
+             ] == MatrixBoard.create_board(4)
+    end
+  end
+
+  describe "rows, columns, diagonals" do
+    test "gets columns" do
+      board = [
```

Comment:
We didn't spend much time on this yesterday, but tests are code too, and the same rules about cleaning up, refactoring, and maintainability should apply to them, too. This board is repeated several times across these tests—I would pull it up into a module attribute.

A rule I like to follow is "three strikes": duplicate something twice times, then refactor to remove the duplication the third time you find yourself copty-pasting.